File: python/containsDuplicate.py
# Time -> 0(N) Space -> 0(N)
def contains(nums: list[int]) -> bool:
    duplicate = {}
    for num in nums:
        exists = duplicate.get(num, False)
        if exists:
            return True
        duplicate[num] = True
    return False


print(contains([0, 2, 3, 4, 4]))
print(contains([0, 2, 3, 4, 5]))

File: go/ContainsDuplicate/ContainsDuplicate.go
package main

import "fmt"

// Time -> 0(N) Space -> 0(N)
func contains(nums []int) bool {
	duplicate := make(map[int]bool, 0)
	for _, num := range nums {
		if _, ok := duplicate[num]; ok {
			return true
		}
		duplicate[num] = true
	}
	return false
}

func main() {
	fmt.Println(contains([]int{0, 2, 3, 4, 4}))
	fmt.Println(contains([]int{0, 2, 3, 4, 5}))
}
